homeassistant/components/mysensors/device.py

```diff
@@ -84,6 +86,25 @@ def device_state_attributes(self):
             else:
                 self._values[value_type] = value
 
+    async def _async_update_callback(self):
+        """Update the device."""
+        raise NotImplementedError
+
+    @callback
+    def async_update_callback(self):
+        """Update the device after delay."""
+        if self._update_scheduled:
+            return
+
+        @callback
+        def update():
+            """Perform update."""
+            self.hass.async_create_task(self._async_update_callback())
```

Comment:
Why not do something like this:

```python
try:
    await self._async_update_callback()
except Exception:
    _LOGGER.exception("Error updating self.entity_id)
finall:
    self._update_scheduled = False
```

That way we can:
 - avoid scheduling updates while an update is in progress
 - able to attribute exceptions to the entity that caused it. This one is something we have been doing poorly in HA and that's why we get so often random stack traces that we can't attribute anymore to the original entity